src/plots/cartesian/axis_defaults.js

```diff
@@ -207,6 +207,7 @@ function isBoxWithoutPositionCoords(trace, axLetter) {
 
     return (
         Registry.traceIs(trace, 'box') &&
+        trace._fullInput.type === 'box' &&
```

Comment:
See https://github.com/plotly/streambed/pull/3236 for the details on why axis auto type with box traces needs more attention. 

I'm debating whether we should add a fallback here for `_fullInput` ?